server/body-controller.js

```diff
@@ -1,71 +0,0 @@
-let creatures = [
-  {
-    id: 1,
-    name: "Halloween Cat",
-    image_url:
-      "https://www.animalsheltering.org/sites/default/files/images/magazine/black-cat-pumpkins-bays-blog-m146689.jpg",
-    scary: {
-      level: 1
-    }
-  },
-  {
-    id: 2,
-    name: "Ghost",
-    image_url:
-      "http://konknet.com/-/wp-content/uploads/2017/08/a-ghost-story-thymb.jpg",
-    scary: {
-      level: 7
-    }
-  }
-];
-
-let id = 3;
-
-module.exports = {
-  read: (req, res, next) => {
-    console.log("I am in GET");
-    res.status(200).send(creatures);
-    console.log("After Read : ", creatures);
-  },
-  create: (req, res, next) => {
-    console.log("I am in POST", req.body);
-    const { name, image_url, scary } = req.body;
-    const newCreature = {
-      id: id,
-      name,
-      image_url,
-      scary
-    };
-    id++;
-    creatures.push(newCreature);
```

Comment:
I think you'll want to reverse lines 39 & 40, since you're starting your id at 3, you want to push first, then increment. What you have here will still work, but you'll skip adding a creature with an id of 3

Reply:
But newCreature is already created by then.. would newCreature's id change if I id++ once its created?

Reply:
you're correct. objects are references but since this is a primitive value, it shouldn't change 

<img width="353" alt="screen shot 2018-09-27 at 12 03 39 pm" src="https://user-images.githubusercontent.com/13579578/46165247-66eeda80-c24d-11e8-892c-ace1e480c2da.png">